### PR TITLE
Basic example of External Workspace Manager Plugin usage

### PR DESCRIPTION
**Pull request by alexsomai**
CC @oleg-nenashev @martinda

### CODE REVIEW

**Comment by martinda** [comment]
LGTM.

**Comment by rtyler** [comment]
@alexsomai Since this repository is incorporated into jenkins.io and serves as user facing documentation, I would prefer to wait to merge this change until the External Workspace Manager plugin is in the main update center, does that make sense to you?

**Comment by oleg-nenashev** [comment]
@rtyler The release is planned to this week, but yeah there is no urgency with merging

If we could have another example for Workspace cleanup plugin, it would be perfect

File: pipeline-examples/external-workspace-manager/externalWorkspace.groovy [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// allocate a Disk from the Disk Pool defined in the Jenkins global config
+def extWorkspace = exwsAllocate 'diskpool1'
+
+// on a node labeled 'linux', perform code checkout and build the project
+node('linux') {
+    // compute complete workspace path, from current node to the allocated disk
+    exws(extWorkspace) {
```
Comment (oleg-nenashev) [inline pipeline-examples/external-workspace-manager/externalWorkspace.groovy]: IMHO there should be also an example with Workspace cleanup

**Comment by alexsomai** [comment]
@rtyler I do agree with you, this should be merged after the stable release
@oleg-nenashev sure, I can add another example with workspace cleanup

File: pipeline-examples/external-workspace-manager/externalWorkspace.groovy [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            // run tests in the same workspace that the project was built
+            sh 'mvn test'
+        } catch (e) {
+            // if any exception occurs, mark the build as failed
+            currentBuild.result = 'FAILURE'
+            throw e
+        } finally {
+            // perform workspace cleanup only if the build have passed
+            // if the build has failed, the workspace will be kept
+            step([$class: 'WsCleanup', cleanWhenFailure: false])
```
Comment (alexsomai) [inline pipeline-examples/external-workspace-manager/externalWorkspace.groovy]: @oleg-nenashev I've added workspace cleanup to the same script. Do you think I should add it as a separate example (a separate script file)?

**Comment by alexsomai** [comment]
@rtyler the plugin is now available in the main update center.
Also I have addressed @oleg-nenashev's suggestion.

**Pull request merged**